WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt

```diff
@@ -145,23 +145,27 @@ class UploadStarter @Inject constructor(
     private suspend fun upload(site: SiteModel) = coroutineScope {
         try {
             mutex.lock()
-            val posts = postStore.getPostsWithLocalChanges(site)
-
-            posts
+            postStore.getPostsWithLocalChanges(site)
                     .asSequence()
-                    .filter {
-                        val action = uploadActionUseCase.getAutoUploadAction(it, site)
-                        if (action == DO_NOTHING) {
-                            false
-                        } else {
-                            trackAutoUploadAction(action, it.status)
-                            true
-                        }
+                    .map { post ->
+                        val action = uploadActionUseCase.getAutoUploadAction(post, site)
+                        Pair(post, action)
```

Comment:
Love this! I stared at this part the last time I reviewed it because I was sure it can be better. I drew blanks during that time.